content/blog/2020/07/2020-07-03-github-checks-api-plugin-coding-phase-1.adoc

```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: "GitHub Checks API Plugin Project - Coding Phase 1"
+tags:
+- github
+- api
+- plugins
+- developer
+- gsoc
+- gsoc2020
+author: XiongKezhi
+opengraph:
+  image: /images/gsoc/opengraph.png
+---
+
+This blog post is about our coding phase 1 progress on GSoC project: link:/projects/gsoc/2020/projects/github-checks/[GitHub Checks API Plugin].
+
+The GitHub Checks API is a highly customized way to integrate CI tools to make reports for pull-requests (PRs). 
+It allows users to see CI reports on GitHub pages directly.
+
+image:/images/post-images/2020/07-github-checks-api-plugin-coding-phase-1/github-check-run.png[GitHub Checks API]
+
+What's more exciting is that it can leave annotations on specific lines of code, just as the comments people left while reviewing.
+
+image:/images/post-images/2020/07-github-checks-api-plugin-coding-phase-1/github-check-annotations.png[Annotations]
+
+While on Jenkins' side, the source code view provided by link:https://plugins.jenkins.io/warnings-ng/[Warnings Next Generation Plugin] does pretty much the same thing.
+
+image:/images/post-images/2020/07-github-checks-api-plugin-coding-phase-1/source-view.png[Source Code View]
+
+Utilizing such features through GitHub Checks API, it would make Jenkins more convenient to GitHub users.
+
+== Features from Coding Phase 1
+
+In the past month, our team was mostly working on the general checks API and an implementation for GitHub checks API.
+
+.GitHub Checks API Plugin Demo [starts from 50:15]
+video::HQLhakpx5mk[youtube,width=800,height=420]
+
+=== General Checks API
+
+Although the general checks API is developed based on the semantic meaning of GitHub Checks API, we still want to prepare it for similar concepts on other platforms like link:https://docs.gitlab.com/ee/api/commits.html#commit-status[Commit Status API] from GitLab.
+Contributions for implementations on these platforms will be welcomed in the future.
+
+=== GitHub Checks API Implmentation
```

Comment:
Maybe it makes sense to write that checks are already running even if no consumer has been implemented yet.